src/dsp_tools/commands/excel2json/new_lists.py

```diff
@@ -12,33 +12,105 @@
 from dsp_tools.models.exceptions import InputError
 
 
-def _fill_id_and_parent_id_columns(df: pd.DataFrame, preferred_language: str) -> pd.DataFrame:
+def _construct_ids(excel_dfs: dict[str, dict[str, pd.DataFrame]]) -> dict[str, dict[str, pd.DataFrame]]:
+    filled = {
+        filename: {
+            sheet_name: _fill_one_df_auto_id_resolve_duplicates(df, _get_preferred_language(df.columns))
+            for sheet_name, df in sheets.items()
+        }
+        for filename, sheets in excel_dfs.items()
+    }
+    filled = _analyse_resolve_all_excel_duplicates(filled)
+    return _fill_all_excel_parent_id(filled)
+
+
+def _fill_all_excel_parent_id(excel_dfs: dict[str, dict[str, pd.DataFrame]]) -> dict[str, dict[str, pd.DataFrame]]:
+    return {
+        filename: {
+            sheet_name: _fill_one_df_parent_id(df, _get_preferred_language(df.columns))
+            for sheet_name, df in sheets.items()
+        }
+        for filename, sheets in excel_dfs.items()
+    }
+
+
+def _analyse_resolve_all_excel_duplicates(
```

Comment:
This function name is a bit a "heap of words": "analyse resolve" is not a meaningful syntactic construct. I suggest to omit "analyse", because the task that this function ultimately does is: resolve the duplicates. Surely, it first has to analyse them, but this doesn't matter for the result.